code/API_definitions/Traffic_Influence.yaml

```diff
@@ -641,28 +654,29 @@ components:
             - 'error'
             - 'deletion in progress'
             - 'deleted'
-        trafficFilters:
-          type: array
-          items:
-            type: string
-            description: Indicates the packet filters of the IP flow. The source
-             IP is not required. It is retrived by the platoform according to
-             the TI API request. If no Traffic Influece resourse is created with
-             a specific Device, any IP will be routed to the Application. If a
-             Traffic Influece resource exists with a specific Device configured,
-             only the related IPs will be routed to the local instance of the
-             Application. The destination IP is not required,it is already known
-             by the platform thanks to the appId and appInstanceId parameters
-             that are used to retive the destination IP. The protocol must be
-             specified and it is identified by a string according to the column
-             “Keyword” as defined by IANA (https://www.iana.org/assignments/\
-             protocol-numbers/protocol-numbers.xhtml), e.g. UDP or TCP.
-             The destination port must be specified.
-            example: TCP 5060
-            minItems: 1
-          minItems: 1
-          description: Identifies IP packet filters. To be used when a the
-           Application needs a traffic flow towards a specific EAS interface
+        sourceTrafficFilters:
+          description: Ports used locally by the device for flows to which
+           the requested traffic influence should apply. If omitted, then the
+           traffic influence will apply to all flows between the device and the
+           specified application server address and ports.
+          type: object
+          properties:
+            sourcePort:
```

Comment:
Shouldnt it have destinationProtocol too in sourceTrafficFilters?

Reply:
I don't think so. the Protocol for the flow is the one defined in destinationTrafficFilter. It should be the same also for the source

Reply:
I agree. So then should we separate out port so that it is explicitly mentioned as a distinct field which associate with both source and destination filters commonly?

Reply:
We could have:

sourceTrafficPort
destinationTrafficPort
trafficProtocol

and we remove trafficFilters

Reply:
The only thing i feel is that if we should include the source or destination IP address or not? If we want to keep IP addresses then trafficFilters still seems fine. If we dont include IP addresses then the above proposal looks fine.

If we chose to keep IP addresses then it could be in similar lines sourceTrafficFilter, destinationTrafficfilter, trafficProtocol?

Reply:
not sure to get the comment :-)

we already have:

- device: with the source IP (public and private) and public Port

I'm proposing to add:

- sourceTrafficPort: the source private Port
- destinationTrafficPort: the destinatin public port
- trafficProtocol: the protocol, that is the same for source and destination

Reply:
With minor change I agree with this,
- Device refers to Source and Destination refers to EAS
- device: represented by the source public IP and public Port (we may not need to use private IP in my opinion)
- sourceTrafficPort: the source (device) public Port
- destinationTrafficPort: the destination public port
- trafficProtocol: the protocol, that is the same for source and destination

Please note that everything is expressed as public from external applications perspective.

Reply:
my understanding was that we had to also provide information about the device PRIVATE port. The source public port is already in Device.

this according to @babunkj , in this comment: https://github.com/camaraproject/EdgeCloud/discussions/230#discussioncomment-9644600

Reply:
we need then more information from @babunkj As in my opinion externally only the public IP and port are visible. Unless it is explicitly specified that the applications hosted in telco edge cloud zone can see both private and public IP:Port.

In general we do not have any visibility how the EAS will see the IP port in device traffic when hosted in edge cloud zone and if different operator will have different model to expose private or public IP of the device it will be difficult to say which one the EAS will see in which deployment. I think we need more feedback from the operator on the possibilities and how well we can address them.

Reply:
Let me first try to (re)state the requirement before giving the comments:
a)	The scenario is: A device has more than one traffic originating from it with the same source IP Address and different ports, but only a specific traffic (identified by a specific port) needs to be influenced and diverted to say an Edge cloud. This requirement needs the identification of the source traffic by its source IP Address and source port. 
An example use case is: when an always-on low resolution video traffic from a device is analysed and a trigger is detected, switch the high resolution video traffic alone to the edge cloud.

b)	In an extended case of the above one:  based on traffic from say Port A, the traffic from a set of ports or all Ports are to be influenced and diverted to an Edge cloud.
An example use case is: when an always-on low resolution video traffic from a device is analysed and a trigger is detected, switch all high resolution video traffic or all video traffic from that device to the edge cloud.


The solution to the above becomes a challenge due to NATing. Giving my comments for IPv4 and IPv6 cases:

IPv4:
-	In the case of IPv4, even though the combination of External IP and External Port will identify the device uniquely, the source port is also required in order to identify the traffic. However, only External IP and External Port are directly visible externally in the case of IPv4.
-	Since it wouldn’t be possible to derive source port directly from the IP headers by the application server, suggesting the following:  

a)	If NAT element supports sharing of IP Address-Port mapping, the source port can be fetched by the operator platform (not available to application server)
b)	When the source port(s) is known to the application server (indirectly), use that info in the source traffic filter. 
c)	When the source port is omitted, the traffic influence is to be applied to all traffic for the session from the device (as provided in the current version)

IPv6:
For IPv6, I suppose, the device port (source port) could be visible to the external server and can be used in the source traffic filter. Not sure how operator networks handle NAT for IPv6.

Another comment is that I feel it is better to keep the sourceTrafficFilters and destinationTrafficFilters attributes as defined in the version 0.9.5-wip. They indicate the purpose. The current structure is also more amenable for future enhancements. 

We could discuss this over a call if that helps.